backend/migrations/versions/5cdcf6cc1d73_create_and_backfill_dataset_table.py

```diff
@@ -0,0 +1,374 @@
+    )
+    op.add_column(
+        's3_dataset',
+        sa.Column('stewards', postgresql.ARRAY(sa.String()), nullable=True),
+    )
+    op.add_column(
+        's3_dataset',
+        sa.Column('SamlAdminGroupName', sa.String(), nullable=True),
+    )
+    op.add_column(
+        's3_dataset',
+        sa.Column('autoApprovalEnabled', sa.Boolean(), nullable=True),  # needs to be updated after backfilling
+    )
+    op.add_column(
+        's3_dataset',
+        sa.Column('datasetType', sa.String(), nullable=True),  # needs to be updated after backfilling
+    )
+    op.add_column(
+        's3_dataset',
+        sa.Column('imported', sa.Boolean(), nullable=True),  # needs to be updated after backfilling
+    )
+    # Fill s3_dataset table with data from dataset table
+    print('Backfill s3_dataset with dataset data...')
+    # Read s3_datasets table rows
+    conn = op.get_bind()
+    session = orm.Session(bind=conn)
+    generic_datasets = {dataset.datasetUri: dataset for dataset in session.query(DatasetBase).all()}
+    s3_datasets = session.query(S3Dataset).all()
+    for dataset in s3_datasets:
+        dataset.label = generic_datasets[dataset.datasetUri].label
+        dataset.name = generic_datasets[dataset.datasetUri].name
+        dataset.owner = generic_datasets[dataset.datasetUri].owner
+        dataset.created = generic_datasets[dataset.datasetUri].created
+        dataset.updated = generic_datasets[dataset.datasetUri].updated
+        dataset.deleted = generic_datasets[dataset.datasetUri].deleted
+        dataset.description = generic_datasets[dataset.datasetUri].description
+        dataset.environmentUri = generic_datasets[dataset.datasetUri].environmentUri
+        dataset.organizationUri = generic_datasets[dataset.datasetUri].organizationUri
+        dataset.region = generic_datasets[dataset.datasetUri].region
+        dataset.AwsAccountId = generic_datasets[dataset.datasetUri].AwsAccountId
+        dataset.language = generic_datasets[dataset.datasetUri].language
+        dataset.topics = generic_datasets[dataset.datasetUri].topics
+        dataset.confidentiality = generic_datasets[dataset.datasetUri].confidentiality
+        dataset.tags = generic_datasets[dataset.datasetUri].tags
+        dataset.businessOwnerEmail = generic_datasets[dataset.datasetUri].businessOwnerEmail
+        dataset.businessOwnerDelegationEmails = generic_datasets[dataset.datasetUri].businessOwnerDelegationEmails
+        dataset.stewards = generic_datasets[dataset.datasetUri].stewards
+        dataset.SamlAdminGroupName = generic_datasets[dataset.datasetUri].SamlAdminGroupName
+        dataset.autoApprovalEnabled = generic_datasets[dataset.datasetUri].autoApprovalEnabled
+        dataset.datasetType = generic_datasets[dataset.datasetUri].datasetType
+        dataset.imported = generic_datasets[dataset.datasetUri].imported
+    session.commit()
+    session.close()
+
+    # Update non-nullable columns
```

Comment:
On downgrade I am getting an error `column "label" of relation "s3_dataset" contains null values` even after we set the `label` value in the above for loop

Reply:
I will re-test with the changes from PR review, see the results below